core/src/main/java/com/linecorp/armeria/common/logging/TextLogFormatter.java

```diff
@@ -188,6 +192,32 @@ public String formatRequest(RequestOnlyLog log) {
         }
     }
 
+    private static void maybeAppendConnectionTimings(@Nullable ClientConnectionTimings timings,
+                                                     StringBuilder buf) {
+        if (timings == null) {
+            return;
+        }
+        buf.append(", {ClientConnection total=");
+        TextFormatter.appendElapsed(buf, timings.connectionAcquisitionDurationNanos());
+        if (timings.dnsResolutionDurationNanos() >= 0) {
+            buf.append(", dns=");
+            TextFormatter.appendElapsed(buf, timings.dnsResolutionDurationNanos());
+        }
+        if (timings.pendingAcquisitionDurationNanos() >= 0) {
+            buf.append(", pending=");
+            TextFormatter.appendElapsed(buf, timings.pendingAcquisitionDurationNanos());
+        }
+        if (timings.socketConnectDurationNanos() >= 0) {
+            buf.append(", socket=");
+            TextFormatter.appendElapsed(buf, timings.socketConnectDurationNanos());
+        }
+        if (timings.tlsHandshakeDurationNanos() >= 0) {
+            buf.append(", tls=");
+            TextFormatter.appendElapsed(buf, timings.tlsHandshakeDurationNanos());
+        }
+        buf.append('}');
```

Comment:
How about specifying the values in the order in which they occurred?
`pending, dns, socket, tls`

Reply:
I thought it was already in that order (`dns, pending, socket, tls`)

dns
https://github.com/line/armeria/blob/3117ad849bf525feda43f44b8f71a7a9109ae2cf/core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java#L112

pending
https://github.com/line/armeria/blob/3117ad849bf525feda43f44b8f71a7a9109ae2cf/core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java#L333

Let me know if I'm misunderstanding though

Reply:
Oops, sorry. 😓